library/common/main_interface.h

```diff
@@ -91,6 +91,13 @@ envoy_engine_t init_engine();
  */
 envoy_status_t set_preferred_network(envoy_network_t network);
 
+/**
+ * Flush the stats sinks outside of a flushing interval.
+ * Note: stats flushing may not be synchronous.
+ * Therefore, this function may return prior to flushing taking place.
+ */
```

Comment:
Might be worth to also document that flushing before the engine has initialized results in no flush.